contracts/REXMarket.sol

```diff
@@ -696,6 +684,24 @@ abstract contract REXMarket is Ownable, SuperAppBase, Initializable {
       return _totalUnitsApproved + _totalUnitsPending > 0 && _balance > 0;
     }
 
+    function _onlyScalable(ISuperToken _superToken, int96 _flowRate) internal virtual {
+      // Enforce speed limit on flowRate
+      require(uint128(uint(int(_flowRate))) % (market.outputPools[market.outputPoolIndicies[_superToken]].shareScaler * 1e3) == 0, "notScalable");
+    }
+
+    function _registerReferral(bytes memory _ctx, address _shareholder) internal {
+      require(referrals.addressToAffiliate(_shareholder) == 0, "noAffiliates");
```

Comment:
Also this line was added, which restricts affiliates from streaming, it looks like if an affiliate is streaming, the accounting for IDA shares is disrupted

Reply:
The rest of the additions in this block come as a result of moving these methods to where they should be in the contract